Policy clarification on managing legacy features (#18518)
nvaccess/nvda: projectDocs/issues/triage.md

Some of the migrated issues have comments that indicate an attachment should be available, but it is not.
These attachments have been lost.

## Deprecated features

> Our core development focus is on modern technologies (like UIA) to ensure the project's long-term health and compatibility.

Reviewer: Markdown lint rules encourage new lines after headers

```suggestion

Our core development focus is on modern technologies (like UIA) to ensure the project's long-term health and compatibility.
```

However, we recognise that some legacy features or settings remain in NVDA out of necessity for specific user workflows.
This creates a special class of issues that require a distinct triage process.
Our goal is to be transparent about our priorities without dismissing the real-world needs of our users.

> When triaging an issue related to a feature that involves a deprecated technology or is otherwise no longer under active development by the core team:

Reviewer 2: ```suggestion
When triaging an issue related to a feature that involves a deprecated technology or is otherwise no longer under active development by the core team:

```

- Identify and label: The first step is to identify if the issue is tied to a legacy component (eg the legacy Word object model).

> Apply a specific label for that component (eg `word-object-model`).

Reviewer: this label doesn't exist

> Apply a specific label for that component (eg `word-object-model`).

Author: Did we decide on a generic label to use for these kinds of issues?

Some of the migrated issues have comments that indicate an attachment should be available, but it is not.
These attachments have been lost.

## Deprecated / legacy features

Our core development focus is on modern technologies (like UIA) to ensure the project's long-term health and compatibility.

However, we recognise that some legacy features or settings remain in NVDA out of necessity for specific user workflows.
This creates a special class of issues that require a distinct triage process.
Our goal is to be transparent about our priorities without dismissing the real-world needs of our users.

When triaging an issue related to a feature that involves a deprecated technology or is otherwise no longer under active development by the core team:

- Migration blockers: Ensure that there is information in the issue on what prevents the user from using the modern alternative (e.g. UIA in Word).
If the user identifies a specific gap or bug in the modern technology (e.g. "I can't use UIA because it doesn't read table headers correctly"), a new issue must be created to track that specific blocker and linked to the legacy issue.

> If there a viable, fully-functional path forward for the user today, then the ticket will be closed.

Author: @CyrilleB79 @SaschaCowley that all sounds reasonable.

By refraining from closing or consolidating staff tickets, we can ensure that the NV Access team maintains control over their internal workflow and prioritisation, while still benefiting from the valuable insights and contributions of the community.

> ## Handling disagreements

Reviewer: ```suggestion
## Handling disagreements

```

> If a strong disagreement arises over the status of an issue (eg its priority, whether it should be closed), do not engage in repeatedly changing the issue's state (eg a "reopen/close war").

Reviewer: I think this is better merged into "Extra permissions for triage" as it's really only relevant to those with  extra permissions to triage

Reviewer: Otherwise

```suggestion
If a strong disagreement arises over the status of an issue (e.g. its priority, whether it should be closed), do not engage in repeatedly changing the issue's state (e.g. a "reopen/close war").
```

Author: This has now been moved.

Candidates will be considered on a case by case basis.

### Handling disagreements

If conflict arises between how best to triage an issue, please defer to NV Access and keep the issue in an open, untriaged state i.e. with the labels "needs triage" and "blocked/needs-product-decision".
Please do not close feature requests on the grounds of rejecting a feature.
Allow the community to discuss the proposed feature and NV Access to make the decision on accepting it.

If a strong disagreement arises over the status of an issue (e.g. its priority, whether it should be closed), do not engage in repeatedly changing the issue's state (e.g. a "reopen/close war").
This is counter-productive and creates a negative atmosphere.

The correct procedure is to:
1.  **Stop:** Leave the issue in its current state.
2.  **Label:** Apply the `blocked/needs-product-decision` label.
3.  **Escalate:** Add a comment explaining the disagreement and tag a relevant NV Access staff member to make a final decision.

This moves the debate from a public power struggle to a productive internal resolution.

> Triagers with extra permissions are expected to lead by example in de-escalating conflict. Repeatedly changing an issue's state in opposition to another triager or staff member is a misuse of permissions and may lead to them being revoked.

Reviewer: ```suggestion
Triagers with extra permissions are expected to lead by example in de-escalating conflict.
Repeatedly changing an issue's state in opposition to another triager or staff member is a misuse of permissions and may lead to them being revoked.
```
